mlflow/gateway/client.py

```diff
@@ -0,0 +1,135 @@
+import json
+import logging
+from typing import Optional
+
+from mlflow.exceptions import MlflowException
+from mlflow.gateway.config import Route
+from mlflow.gateway.constants import (
+    MLFLOW_GATEWAY_ROUTE_BASE,
+    MLFLOW_GATEWAY_HEALTH_ENDPOINT,
+    MLFLOW_GATEWAY_DATABRICKS_ROUTE_PREFIX,
+)
+from mlflow.gateway.utils import (
+    _resolve_gateway_uri,
+    _merge_uri_paths,
+)
+from mlflow.tracking._tracking_service.utils import _get_default_host_creds
+from mlflow.utils.annotations import experimental
+from mlflow.utils.databricks_utils import get_databricks_host_creds
+from mlflow.utils.rest_utils import MlflowHostCreds, http_request, augmented_raise_for_status
+from mlflow.utils.uri import get_uri_scheme
+
+
+_logger = logging.getLogger(__name__)
+
+
+@experimental
+class MlflowGatewayClient:
+    """
+    Client for interacting with the MLflow Gateway API.
+
+    :param gateway_uri: Optional URI of the gateway. If not provided, attempts to resolve from
+    first the stored result of `set_gateway_uri()`, then the  environment variable
+    `MLFLOW_GATEWAY_URI`.
+    """
+
+    def __init__(self, gateway_uri: Optional[str] = None):
+        self._gateway_uri = _resolve_gateway_uri(gateway_uri)
```

Comment:
Can we just do:

```suggestion
        self._gateway_uri = gateway_uri or get_gateway_uri()
```

?

Reply:
good point. Moved to the client constructor.